src/app.js

```diff
@@ -44,12 +46,17 @@ const AppFactory = (args) => {
     res.redirect('/signup');
   });
 
-  // TODO: refactor and make new views directory?
   app.get('/signup', (req, res) => {
     res.send(signupTemplate({ req }));
   });
 
-  app.post('/signup', async (req, res) => {
+  app.post('/signup', [parseEmail, parsePassword, parsePasswordConfirmation], async (req, res) => {
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty()) {
+      return res.send(signupTemplate({ req, errors }));
+    }
+
     return createUserController.execute(req, res);
```

Comment:
for sure, the more we put into the route handler, the less of a reason there is to have the controllers. In that case, we'd keep the routes in this file, delete controllers, have the routes call the service methods. Then you have HTTP (routes), business (services), and external db clients (repo). By contrast, the _HTTP (controllers), business (services), and external db clients (repo)_ pattern works better when you don't really need to think about code in the routes because they immediately hand off everything to the controller.

This file might get bigger with a refactor at the HTTP layer, but, yeah, if we're not parsing in the controllers, then they start to lose their importance.